src/models/cart-model.js

```diff
@@ -211,7 +211,9 @@ const CartModel = BaseModel.extend({
     const newLength = newLineItems.length;
 
     if (newLength < oldLength) {
-      this.attrs.line_items = newLineItems;
+      this.attrs.line_items = newLineItems.map(item => {
```

Comment:
I'm not a fan of 'as needed' because then the moment you need to grow the method, you need to start decorating it with more stuff. I'm a big fan of consistent, explicit conventions.